### PR TITLE
ci: Add initial GitHub Actions config

### PR DESCRIPTION
**Pull request by hawkw**
This is just a fairly simple `cargo check` + `rustfmt` workflow for now.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
```
Comment (olix0r) [inline .github/workflows/rust.yml]: prefer to use shas to tags, especially for actions not owned by github

**Review by olix0r — APPROVED** [review]
lgtm. would prefer not relying on tags for third-party actions

File: .github/workflows/rust.yml (hunk 1 above)
Comment (hawkw) [reply .github/workflows/rust.yml]: TIL that you can do that! AFAICT, this means that we won't get minor bugfix patches to the action unless we explicitly update the SHA — what's the reason to prefer this?

**Review by kleimkuhler — APPROVED** [review]
👍

File: .github/workflows/rust.yml (hunk 1 above)
Comment (olix0r) [reply .github/workflows/rust.yml]: I don't like CI changing underneath us without being able to opt-in to change. It means things can just break and we won't know why.

More generally, tags open up an angle for someone to be able to run arbitrary code in our CI (i.e. if a maintainer loses control of their credentials).

Comment (hawkw) [reply .github/workflows/rust.yml]: That's fair. I'll change this to use SHAs.

**Pull request merged**